learning/tour-of-beam/frontend/lib/pages/tour/state.dart

```diff
@@ -115,17 +134,20 @@ class TourNotifier extends ChangeNotifier with PageStateMixin<void> {
     if (content == null) {
       return;
     }
-
     final taskSnippetId = content.taskSnippetId;
-    if (taskSnippetId == null) {
+    await _setPlaygroundSnippet(taskSnippetId);
```

Comment:
1. What if we are in the 'solution' mode?
2. I suggest making UnitContentModel equatable. Otherwise if the same model is re-fetched from the server, the `==` above is false and we lose changes here.

Reply:
1. Is solution mode supposed to be persisted after switching between units?

Reply:
No.

Reply:
`_setCurrentUnitContent` is called in `_onUnitChanged` and `_showSolution` doesn't affect it.

Reply:
I mean we should reset `_showSolution` here. Otherwise we may have `_showSolution == true` but showing a snippet for a task and not a solution.